packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,194 @@
+
+enum relayState {
+  Pending,
+  SpedUp,
+  Disputed,
+  Finalized,
+}
+
+enum isRelayValid {
+  Yes,
+  No,
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  bridgeAdminAddress: string;
+}
+
+interface Relay {
+  bridgePoolAddress: string;
+  quoteTimestamp: number;
+  depositId: number;
+  sender: string;
+  slowRelayer: string;
+  disputedSlowRelayers: string[];
+  instantRelayer: string;
+  depositTimestamp: number;
+  recipient: string;
+  l1Token: string;
+  amount: string;
+  slowRelayFeePct: string;
+  instantRelayFeePct: string;
+  realizedLpFeePct: string;
+  priceRequestAncillaryDataHash: string;
+  depositHash: string;
+  depositContract: string;
+  relayState: relayState;
+  relayTimestamp: number;
+}
+
+interface Deposit {
+  depositId: number;
+  timestamp: number;
+  sender: string;
+  recipient: string;
+  l1Token: string;
+  amount: string;
+  slowRelayFeePct: string;
+  instantRelayFeePct: string;
+  quoteTimestamp: number;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond True or False
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly bridgeAdminAddress: string;
+  private l1Client: InsuredBridgeL1Client | null = null;
+  private l2Client: InsuredBridgeL2Client | null = null;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {String} bridgeAdminAddress Address of BridgeAdmin contract on L1.
+   * @param
+   */
+  constructor({ logger, web3, bridgeAdminAddress }: Params) {
+    super();
+
+    assert(isAddress(bridgeAdminAddress), "bridgeAdminAddress required");
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.bridgeAdminAddress = bridgeAdminAddress;
+  }
+
+  public async getHistoricalPrice(time: number | string): Promise<BN> {
+    // Grab price request for a timestamp and parse the ancillary data for the request.
+    const matchedRelays = this.relays.filter((relay: Relay) => relay.relayTimestamp === time);
+    if (matchedRelays.length > 1) throw new Error("TODO: Handle multiple relays for same price request timestamp");
+    if (matchedRelays.length === 0) throw new Error("No price request for time");
+    const relay = matchedRelays[0];
+
+    // TODO: Not 100% sure we should reconstruct the relay ancillary data using deposit/relay data from the L2/L1
```

Comment:
@chrismaree @mrice32 want to draw your attention here. I originally thought we needed to parse the price request's ancillary data to determine relay params, but alternatively we can just use the clients data. WDYT?